src/core/symbol/SymbolTransaction.ts

```diff
@@ -0,0 +1,39 @@
+import { Converter, KeyPair, SymbolNetwork, SymbolTransactionUtils } from '@core';
+import { TransactionBuilder, TransactionHelper } from 'catbuffer-typescript';
+
+export interface SignedTransaction {
+    payload: Uint8Array;
+    signature: Uint8Array;
+    transactionHash: string;
+}
+
+export interface TransactionSigner {
+    signTransaction(payload: Uint8Array, generationHashBytes: Uint8Array): Promise<SignedTransaction>;
```

Comment:
Why promise? 🤔

Reply:
Mobed to sync operation as explained below :)